sdk/provisioning/Azure.Provisioning.CloudMachine/api/Azure.Provisioning.CloudMachine.netstandard2.0.cs

```diff
@@ -46,30 +44,34 @@ namespace Azure.Core
     public partial class ClientCache
     {
         public ClientCache() { }
-        public T Get<T>(string id, System.Func<T> value) where T : class { throw null; }
+        public T Get<T>(System.Func<T> value, string? id = null) where T : class { throw null; }
+    }
+    public enum ClientConnectionKind
+    {
+        EntraId = 0,
+        ApiKey = 1,
+        OutOfBand = 2,
```

Comment:
You might prefer something like `Custom` or `ServiceSpecific` or `Bespoke` or even `Other`?

Reply:
It's basically for cases where something else supplies the creds, e.g. ServiceBusClient provides creds for ServiceBusSender

Reply:
I don't know if `OutOfBand` is the right way to communicate that?  It makes me think more about package releases.